core/src/main/java/org/javarosa/core/model/actions/Action.java

```diff
@@ -37,9 +46,7 @@ public Action(String name) {
      * @return TreeReference targeted by the action or null if the action
      * wasn't completed.
      */
-    public TreeReference processAction(FormDef model, TreeReference context) {
-        return null;
-    }
+    public abstract TreeReference processAction(FormDef model, TreeReference context);
```

Comment:
We used to need Action to be non-abstract because we were serializing it directly, if I recall correctly.